src/problem.ts

```diff
@@ -1 +1,61 @@
-export class Problem {};
\ No newline at end of file
+import { CONTENT_TYPE, Problem } from "types";
+export class AsyncApiProblem extends Error implements Problem {
+
+    public "Content-Type" = CONTENT_TYPE;
+    public type: string;
+    public title: string;
+    public errorUrlPrefix: string;
+    public status: number;
+    public instance?: string;
+    public skipFields?: string[];
+    public detail?: string;
+    public parsedJSON?: object | undefined;
+    public validationErrors?: object[] | undefined;
+    public location?: object[] | undefined;
+    public refs?: object[] | undefined;
```

Comment:
`status`, `parsedJSON`, `validationErrors`, `location`, `refs` should be here. You copied it from existing implementation but this library should be generics as possible.

Also we will ride most of them (I mean mentioned props) from `@asyncapi/parser`, because we will integrate spectral with our parser, so we won't need it anymore.

I have only problem with `status`, because we need it for `@asyncapi/server-api`. Maybe rather `status` on root we should have `http` field with `status` and `contentType` like:

```ts
http: {
  status: number,
  contentType: 'application/problem+json',
}
```

wdyt?

Reply:
Remove `contentType` completely. It will be added on the server side. There is no point in keeping it in the problem side.

Reply:
Sure, I think, removing content-type willl be clean code.
Making `http` look like

```ts
    http: {
        status: number,          // Status Code
        [key: string]: any,
    }
```